### PR TITLE
[Sprint 3] feature login logic #1

### PR DESCRIPTION
**Pull request by HyperionsThs**
Effect : N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/fbook/data/source/remote/api/request/SignInRequest.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  @Expose
+  @SerializedName("email")
+  var email: String? = null
+  @Expose
+  @SerializedName("password")
+  var password: String? = null
+  @SerializedName("refresh_token")
+  @Expose
+  var refreshToken: String? = null
+}
```
Comment (NguyenTienVinh) [inline app/src/main/java/com/framgia/fbook/data/source/remote/api/request/SignInRequest.kt]: enter

File: app/src/main/java/com/framgia/fbook/data/source/remote/api/response/SignInResponse.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  inner class SignInData {
+    @SerializedName("access_token")
+    @Expose
+    var accessToken: String? = null
+    @SerializedName("refresh_token")
+    @Expose
+    var refreshToken: String? = null
+  }
+}
```
Comment (NguyenTienVinh) [inline app/src/main/java/com/framgia/fbook/data/source/remote/api/response/SignInResponse.kt]: enter

File: app/src/main/java/com/framgia/fbook/data/source/remote/api/request/SignInRequest.kt (hunk 1 above)
Comment (HyperionsThs) [reply app/src/main/java/com/framgia/fbook/data/source/remote/api/request/SignInRequest.kt]: entered

File: app/src/main/java/com/framgia/fbook/data/source/remote/api/response/SignInResponse.kt (hunk 2 above)
Comment (HyperionsThs) [reply app/src/main/java/com/framgia/fbook/data/source/remote/api/response/SignInResponse.kt]: entered

File: app/src/main/java/com/framgia/fbook/data/source/remote/api/service/NameApi.kt [hunk 3]
```diff
@@ -2,12 +2,12 @@ package com.framgia.fbook.data.source.remote.api.service
...
 import com.framgia.fbook.data.model.User
+import com.framgia.fbook.data.source.remote.api.request.SignInRequest
 import com.framgia.fbook.data.source.remote.api.response.BaseResponse
 import com.framgia.fbook.data.source.remote.api.response.SearchUserResponse
+import com.framgia.fbook.data.source.remote.api.response.SignInResponse
 import io.reactivex.Single
-import retrofit2.http.GET
-import retrofit2.http.Path
-import retrofit2.http.Query
+import retrofit2.http.*
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/data/source/remote/api/service/NameApi.kt]: avoid ``import .*``
Pls keep old code

File: app/src/main/java/com/framgia/fbook/screen/login/LoginPresenter.kt [hunk 4]
```diff
@@ -1,24 +1,41 @@
...
 
   override fun setViewModel(viewModel: LoginContract.ViewModel) {
     mViewModel = viewModel
   }
 
+  override fun login(signInRequest: SignInRequest) {
+    val disposable: Disposable = mUserRepository.login(signInRequest)
+        .subscribeOn(mSchedulerProvider.io())
+        .observeOn(mSchedulerProvider.ui())
+        .subscribe({ t1 -> mViewModel?.onLoginSuccess(t1) },
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/login/LoginPresenter.kt]: pls rename ``t1`` -> ``signInRespone``

File: app/src/main/java/com/framgia/fbook/screen/login/LoginPresenter.kt [hunk 5]
```diff
@@ -1,24 +1,41 @@
...
+            { t -> mViewModel?.onError(t as BaseException) })
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/login/LoginPresenter.kt]: pls rename ``t`` -> ``e``

**Pull request merged**